### PR TITLE
Adds role for deploying Open Data Hub

### PR DESCRIPTION
**Pull request by jdob**
This role installs the community version of the ODH operator using OLM.
It then uses the operator to deploy ODH itself.

##### ISSUE TYPE

- New role Pull Request

##### COMPONENT NAME

ocp4-workload-open-data-hub-olm

##### ADDITIONAL INFORMATION
This role installs the community version of the Open Data Hub Operator using OLM. Once it's installed, a new project is created and Open Data Hub is deployed to it.

```paste below

```

### CODE REVIEW

File: ansible/roles/ocp4-workload-open-data-hub-olm/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+become_override: False
+ocp_username: opentlc-mgr
+silent: False
+
+odh_namespace: "opendatahub"
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-open-data-hub-olm/defaults/main.yml:6]: Please change variable names (except the first three) to use the `ocp4_workload_open_data_hub_olm_` prefix. For example `ocp4_workload_open_data_hub_olm_namespace`

File: ansible/roles/ocp4-workload-open-data-hub-olm/readme.adoc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ansible-playbook -i ${TARGET_HOST}, ./configs/ocp-workloads/ocp-workload.yml \
+    -e"ansible_ssh_private_key_file=~/.ssh/id_rsa" \
+    -e"ansible_user=ec2-user" \
+    -e"ocp_username=${OCP_USERNAME}" \
+    -e"ocp_workload=${WORKLOAD}" \
+    -e"user_count=50" \
+    -e"guid=${GUID}" \
+    -e"ACTION=remove"
+----
+
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-open-data-hub-olm/readme.adoc:72]: You can delete everything past here.

File: ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    definition: "{{ lookup('template', 'subscription.yaml.j2' ) }}"
+
+- name: Wait for the CRD to be available
+  k8s_facts:
+    api_version: "apiextensions.k8s.io/v1beta1"
+    kind: CustomResourceDefinition
+    name: "{{ item }}"
+  loop:
+    - kfdefs.kfdef.apps.kubeflow.org
+  register: crds
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml:18]: We usually use the `r_` prefix on variable set via register. So this would be `r_crds`.

File: ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- name: Create Namespace to deploy Open Data Hub into
+  k8s:
+    state: present
+    definition: "{{ lookup('template', 'namespace.yaml.j2') }}"
+
+- name: Deploy Open Data Hub
+  k8s:
+    state: present
+    namespace: "{{ odh_namespace }}"
+    definition: "{{ lookup('template', 'opendatahub-cr.yaml.j2') }}"
```
Comment (wkulhanek) [inline ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml:32]: It would be great to add logic to make sure it deployed successfully. Examples in the roles_ocp_workload / logging or serverless roles.

File: ansible/roles/ocp4-workload-open-data-hub-olm/readme.adoc [hunk 5]
```diff
@@ -0,0 +1,135 @@
+= ocp4-workload-open-data-hub-olm - Deploy Open Data Hub using the community operator included in OLM
+
+== Role overview
+
+* This role deploys Open Data Hub. It consists of the following playbooks:
+** Playbook: link:./tasks/pre_workload.yml[pre_workload.yml] - Sets up an
```
Comment (jkupferer) [inline ansible/roles/ocp4-workload-open-data-hub-olm/readme.adoc:6]: This is a tasks file, not a playbook.

Comment (jdob) [reply ansible/roles/ocp4-workload-open-data-hub-olm/readme.adoc:6]: I don't mind changing this, but I copy/pasted it from other roles which all use the same boilerplate text. For example, the ocp4_workload_piplines, ocp4_workload_logging, and ocp4_workload_serverless roles that wkulhanek suggested I follow all include this stanza, along with all of the others I randomly checked.

I'll change it if you're going to block the PR because of it, but personally, I'm more for consistency and then to address this globally for all of our readmes.

**Comment by jdob** [comment]
" And rename the role to use all underscores instead of dashes in the name."

I'm not opposed to changing this, but nearly all of the other roles under ansible/roles use hyphens and it feels weird to break that convention.

This may point to the larger issue that I should be putting this under ansible/roles_ocp_workloads instead, so I'm hoping you can let me know if you want me to move this under there (or for that matter, what the difference is between "roles" and "roles_ocp_workloads").

If you don't want me to move it out of "roles", can you let me know if you still want me to change to underscores? Like I said, I don't mind changing it, it just feels awkward to have this one role use underscores where dozens of others in that directory use hyphens.

File: ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml (hunk 3 above)
Comment (jdob) [reply ansible/roles/ocp4-workload-open-data-hub-olm/tasks/workload.yml:18]: Done. Out of curiosity, is there a reason for this? I'm not a huge Ansible guy and I'm curious if that's a standard convention or just in this project.

**Comment by wkulhanek** [comment]
@jdob Copy/paste from chat (for posterity. :-) )

So all roles that go into the roles_ocp_workloads should follow those conventions. Along with variable names that use the role name as prefix and other guidelines
The idea behind the roles_ocp_workloads directory was to start fresh for some workloads while not breaking existing configs. E.g. you'll see the same workloads in both - except in the workloads directory they are updated (and documented) as now recommended. At the same time we don't have to go over hundreds of configs updating those...
So bottom line: If it's a role that you are willing to support and that conforms to all published conventions you can most certainly submit the PR to go into the roles_workloads directory.
If it's a one off for some demo / summit /rhte lab that's never being used after..... throw it into just 'roles'
